packages/vats/test/test-vat-bank.js

```diff
@@ -0,0 +1,127 @@
+import { buildRootObject } from '../src/vat-bank';
+
+test('communication', async t => {
+  t.plan(24);
+  const bankVat = E(buildRootObject)();
+
+  /** @type {undefined | { fromBridge: (srcID: string, obj: any) => void }} */
+  let bankHandler;
+
+  /** @type {import('../src/bridge').BridgeManager} */
+  const bridgeMgr = Far('fakeBridgeManager', {
+    register(srcID, handler) {
+      t.is(srcID, 'bank');
+      t.assert(handler);
+      bankHandler = handler;
+    },
+    toBridge(dstID, obj) {
+      t.is(dstID, 'bank');
+      let ret;
+      switch (obj.type) {
+        case 'VPURSE_GET_BALANCE': {
+          const { address, denom, type: _type, ...rest } = obj;
+          t.is(address, 'agoricfoo');
+          t.is(denom, 'ubld');
+          t.deepEqual(rest, {});
+          ret = '11993';
+          break;
+        }
+
+        case 'VPURSE_GIVE': {
+          const { amount, denom, recipient, type: _type, ...rest } = obj;
+          t.is(recipient, 'agoricfoo');
+          t.is(denom, 'ubld');
+          t.is(amount, '14');
+          t.deepEqual(rest, {});
+          ret = amount;
+          break;
+        }
+
+        case 'VPURSE_GRAB': {
+          const { amount, denom, sender, type: _type, ...rest } = obj;
+          t.is(sender, 'agoricfoo');
+          t.is(denom, 'ubld');
+          t.is(amount, '14');
+          t.deepEqual(rest, {});
+          ret = amount;
+          break;
+        }
+
+        default: {
+          t.is(obj, null);
+        }
+      }
+      return ret;
+    },
+    unregister(srcID) {
+      t.is(srcID, 'bank');
+      t.fail('no expected unregister');
+    },
+  });
+
+  // Create a bank manager.
+  const bankMgr = await E(bankVat).makeBankManager(bridgeMgr);
+  const bank = E(bankMgr).getBankForAddress('agoricfoo');
+
+  const sub = await E(bank).getAssetSubscription();
+  const it = sub[Symbol.asyncIterator]();
+
+  const kit = makeIssuerKit('BLD', MathKind.NAT, harden({ decimalPlaces: 6 }));
+  await t.throwsAsync(() => E(bank).getPurse(kit.brand), {
+    message: /"brand" not found/,
+  });
+
+  /** @type {undefined | IteratorResult<{brand: Brand, issuer: Issuer, proposedName: string}>} */
+  let itResult;
+  const p = it.next().then(r => (itResult = r));
+  t.is(itResult, undefined);
+  await E(bankMgr).addAsset('ubld', 'BLD', 'Staking Tokens', kit);
+  await p;
+  t.is(itResult && itResult.done, false);
+
+  // First balance.
+  const vpurse = await E(bank).getPurse(kit.brand);
+  const bal = await E(vpurse).getCurrentAmount();
+  t.assert(amountMath.isEqual(bal, amountMath.make(11993n, kit.brand)));
+
+  // Deposit.
+  const paymentAmount = amountMath.make(14n, kit.brand);
+  const payment = await E(kit.mint).mintPayment(paymentAmount);
+  const actualPaymentAmount = await E(vpurse).deposit(payment, paymentAmount);
+  t.assert(amountMath.isEqual(actualPaymentAmount, paymentAmount));
+
+  // Withdrawal.
+  const payment2 = /** @type {Payment} */ (await E(vpurse).withdraw(
```

Comment:
Did we have to type this explicitly? If so, why?

Reply:
I'll add a comment.  I can't easily type a `VirtualPurse` unless I make it callable only with `E`, in which case I can't unwrap to a sync interface (which `Payment` is).  We can fix this only if the ERTP methods also allow consuming a `Remote<Payment>` instead of just `Payment`.  That typing has not yet been done.